### ISSUE TITLE
AP-731: POST: GroupChatManager - Add access control

### ISSUE THREAD

**Issue by stavares843** [Satellite-im/Core-PWA#1394]
Add the write and read filter functions during the collection creation and update.

On the group creation load standard rules:

only the members of the group can add new messages

only the sender of a specific message can update the message

Add the possibility to update the rules (we will use this functionality when we add new members to the group)

when adding a new member we should be able to update the rules granting permissions to the new  group member

This ticket is blocked by the groupchatmanager one. Meanwhile you can do some tests in codesandbox to better understand how writeValidators and readValidators works for Textile since they are not well documented

**Comment by stavares843** [comment]
> Manuel Tumiati commented in Jira:

We have to rethink the way we manage thread and collections. We discussed that together with Textile guys and they approved our approach. It turned out later that this approach does not allow us to implement secure access control.



My suggestion is to keep it as it is for now and to rewrite our database part  post-alpha

**Comment by stavares843** [comment]
> Sara Tavares commented in Jira:

cc@Manuel Tumiatiany updates about this one?🔨

**Comment by stavares843** [comment]
> Sara Tavares commented in Jira:

considering your comment, should we move this to To Do then?

**Comment by stavares843** [comment]
> Sheldon McGee commented in Jira:

@Sara TavaresI heard it was postponed too so I’m moving it back to todo. Want to add thoughts here@Manuel Tumiatias to why we are postponing?

**Comment by stavares843** [comment]
> Sara Tavares commented in Jira:

thanks@Sheldon McGee

**Comment by stavares843** [comment]
> Manuel Tumiati commented in Jira:

yes, it’s postponed

**Comment by stavares843** [comment]
> Sheldon McGee commented in Jira:

@Manuel TumiatiIs this postponed till after Early Access?
